src/tests/ftest/pool/bad_evict.yaml

```diff
@@ -22,7 +24,7 @@ evicttests:
      badname:
           setname:
              - NULL
-             - FAIL
+             - PASS
```

Comment:
These mux names should probably be updated to reflect the new expected behavior and add the missing valid name case:
```
   connectsetnames: !mux
      goodname:
         setname:
	    - daos_server
            - PASS
      wrongname:
         setname:
            - scott_server
            - PASS
      nullname:
         setname:
            - NULL
            - PASS
```

The code would suggest that only the current `badname` entry has an effect on the test.   Shouldn't the code in bad_evict.py look something more like this?
```
            # trash the pool group value
            savedgroup = pool.group
            if evictset is None:
                pool.group = None
            else:
                pool.group = evictset
```

Reply:
Thanks for the feedbacks and comments, 
After some debug per your suggestion: 
pool.group is  <ctypes.c_char_Array_12 object at 0x7f5e311b2b90>
Current tests mux for pool.group with:
(1)valid pool.group (original object in pool.create for positive test)
(2)Null  (Null object for default pool.group test)

Reply:
Ah, yeah I forgot about the ctype conversion for DaosPool.group.  Updated code snippet:
```
            # trash the pool group value
            savedgroup = pool.group
            if evictset is None:
                pool.group = None
            else:
                pool.group.set_group(evictset)
```

Reply:
resolved.
Please see the new commit#3.